src/org/zaproxy/zap/extension/fuzz/httpfuzzer/processors/tagcreator/RegexTagRule.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2015 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.fuzz.httpfuzzer.processors.tagcreator;
+
+import java.util.regex.Pattern;
+
+public abstract class RegexTagRule implements TagRule {
+    private String regex;
+    private Pattern pattern;
+
+    protected RegexTagRule(String regex) {
+        this.regex = regex;
+        this.pattern = Pattern.compile(regex);
+    }
+
+    public String getRegex(){
+        return regex;
+    }
+
+    protected Pattern getRegexPattern(){
+        return pattern;
+    }
+
+    protected boolean hasRegex() {
+        return regex != null && regex.length() > 0;
```

Comment:
The null check can now be removed (the regex is guaranteed to be non-null at this point of execution as it was already compiled, which would result in an exception if that was the case).